dvc/objects/db/reference.py

```diff
@@ -27,10 +27,15 @@ class ReferenceObjectDB(ObjectDB):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._fs_cache: Dict[tuple, "BaseFileSystem"] = {}
+        self._obj_cache: Dict["HashInfo", "ReferenceHashFile"] = {}
```

Comment:
This thing is actually useful for Trees too, and potentially other obj types. This is to add to our previous discussion about making base odb type-aware @pmrowla